public/javascripts/ers_csv_scripts.js

```diff
@@ -1,122 +1,96 @@
+        return true
+    } else {
+        return fileSize <= MAX_CSV_FILESIZE;
+    }
+}
 
-	function duplicateFileName (fileName) {
-		var duplicateNameCount = 0
-		$(".input-file-name").each(function(index){
-			if ($(this).val() != "") {
-				if (ie<10) {
-					if ($(this).val().substr($(this).val().lastIndexOf("\\")+1, $(this).val().length) == fileName) {
-						duplicateNameCount++;
-						if (duplicateNameCount > 1) $(this).parent("Div").addClass("fileAlert")
-					}
-				} else {
-					if ($(this)[0].files[0].name == fileName) {
-						duplicateNameCount++;
-						if (duplicateNameCount > 1) $(this).parent("Div").addClass("fileAlert")
-					}
-				}
-			}
-		});
-		if (duplicateNameCount > 1) {
-			return true;
-		} else {
-			return false;
-		}
-	}
+function removeFileAlert() {
+    if (document.querySelector(".govuk-form-group--error")) {
+        document.querySelector(".govuk-form-group--error").classList.remove("govuk-form-group--error")
+    }
+}
 
-	function removeFileAlert () {
-		$(".input-file-name").each(function(index){
-			$(this).parent("Div").removeClass("fileAlert")
-		});
-	}
+function showCSVErrorMsg(e, msg) {
+    const button = document.getElementById("check-file-button");
 
-	function showCSVErrorMsg(e, msg) {
-    	if ($("#error-summary").length) {
-    		$("#error-summary").remove();
-    	}
-    	$(".visibility").show();
-    	$("#check-file-button").attr("disabled",true);
-	    $(e).before("<p id='error-summary' class='field-error clear' tabindex'-1' role='alert' aria-labelledby='error-heading'>"+msg+".</p>")
-	    $(".validation-summary-message a").html(msg)
-	    $("#errors").focus();
-	}
+    if (document.getElementById("error-summary")) {
+        document.getElementById("error-summary").remove()
+    }
 
-	function validateFile(fileName, fileSize, e) {
-		// Check file name
-		if (validFileName(fileName)) {
-			// check file name length
-			if (fileName.length <= MAX_FILENAME_LENGTH) {
-				// Check file extn
-				if (getFileNameExtension(fileName) == "csv") {
-					if (csvFileSizeOK(fileSize)) {
-						if (!duplicateFileName(fileName)) {
-							// file ok
-							return true;
-						} else {
-							showCSVErrorMsg(e, GOVUK.getLocalisedContent("csv.already.chosen"));
-					    	errors++;
-							return false;
-						}
-					} else {
-						showCSVErrorMsg(e, GOVUK.getLocalisedContent("csv.file.too.large", [MAX_CSV_FILESIZE/1000000]));
-						errors++;
-						return false;
-					}
-				} else {
-					showCSVErrorMsg(e, GOVUK.getLocalisedContent("csv.file.wrong.type"));
-					errors++;
-					return false;
-				}
-			} else {
-				showCSVErrorMsg(e, GOVUK.getLocalisedContent("csv.file.name.too.long", [MAX_FILENAME_LENGTH]));
-				errors++;
-				return false;
-			}
-		} else {
-			showCSVErrorMsg(e, GOVUK.getLocalisedContent("csv.file.name.invalid.chars"));
-			errors++;
-			return false;
-		}
-	}
+    document.querySelector(".govuk-\\!-display-none").classList.remove("govuk-!-display-none");
```

Comment:
This was a little confusing to me at first. I think you're trying to remove an existing error summary though, if present. I think it would be better to use an id selector?

Reply:
I'm happy to change this to an id selector, I'm guessing the reason you were confused was the double backslash there? exclamation marks need to be cancelled out in JS queries for whatever reason, it seems

Reply:
It was more readability, its not immediately clear what we're selecting and trying to remove the class from.